olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h

```diff
@@ -44,10 +44,30 @@ namespace read {
 class CatalogClientImpl;
 
 /**
- * @brief The CatalogClient class. Marshals Requests and their Results.
+ * @brief The `CatalogClient` class marshals `OLP` requests and their results.
+ *
+ * A client to access data hosted on `OLP` by Http. It
+ * provides a high-level interface for the retrieval of `OLP` catalog data and
+ * configuration.
+ *
+ * The behaviour of the `CatalogClient`can be defined via `OlpClientSettings`.
+ * You can overwrite the default implementation for:
+ *   * the task scheduler. By default all request calls will be performed
+ *     synchronously.
+ *   * the network. You can use either set the default implementation (@see
+ *    olp::client::OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler)
+ *     or pass a customer network implementation can be passed to
+ *     `CatalogClient`.
+ *   * the on-disk cache. By default the `CatalogClient` uses the default
+ *     implementation olp::cache::DefaultCache
  */
 class DATASERVICE_READ_API CatalogClient final {
  public:
+  /**
+   * @brief Constructs a `CatalogClient` instance.
+   * @param catalog Here Resource Name of the `OLP` catalog
```

Comment:
In general, do we call catalogs `'OLP' catalogs` explicitly? Or `Catalog` would suffice?

Reply:
I think, referencing to a generic 'catalog' might be misleading.

Reply:
Ok, might be.